packages/governance/src/tests/test_votes.cairo

```diff
@@ -0,0 +1,309 @@
+
+    state
+}
+
+fn setup_account(public_key: felt252) -> ContractAddress {
+    let mut calldata = array![public_key];
+    utils::declare_and_deploy("DualCaseAccountMock", calldata)
+}
+
+//
+// Common tests for Votes
+//
+
+#[test]
+fn test_get_votes() {
+    let mut state = setup_erc721_votes();
+    start_cheat_caller_address(test_address(), OWNER());
+    // Before delegating, the owner has 0 votes
+    assert_eq!(state.get_votes(OWNER()), 0);
+    state.delegate(OWNER());
+
+    assert_eq!(state.get_votes(OWNER()), 10);
+}
+
+// This test can be improved by using the api of the component
+// to add checkpoints and thus verify the internal state of the component
+// instead of using the trace directly.
+#[test]
+fn test_get_past_votes() {
+    let mut state = setup_erc721_votes();
+    let mut trace = state.Votes_delegate_checkpoints.read(OWNER());
+
+    cheat_block_timestamp_global('ts10');
+
+    trace.push('ts1', 3);
+    trace.push('ts2', 5);
+    trace.push('ts3', 7);
+
+    assert_eq!(state.get_past_votes(OWNER(), 'ts2'), 5);
+    assert_eq!(state.get_past_votes(OWNER(), 'ts5'), 7);
+}
+
+#[test]
+#[should_panic(expected: ('Votes: future Lookup',))]
+fn test_get_past_votes_future_lookup() {
+    let state = setup_erc721_votes();
+    cheat_block_timestamp_global('ts1');
+    state.get_past_votes(OWNER(), 'ts2');
+}
+
+#[test]
+fn test_get_past_total_supply() {
+    let mut state = setup_erc721_votes();
+    let mut trace = state.Votes_total_checkpoints.read();
+
+    cheat_block_timestamp_global('ts10');
+    trace.push('ts1', 3);
+    trace.push('ts2', 5);
+    trace.push('ts3', 7);
+
+    assert_eq!(state.get_past_total_supply('ts2'), 5);
+    assert_eq!(state.get_past_total_supply('ts5'), 7);
+}
+
+#[test]
+#[should_panic(expected: ('Votes: future Lookup',))]
+fn test_get_past_total_supply_future_lookup() {
+    let state = setup_erc721_votes();
+    cheat_block_timestamp_global('ts1');
+    state.get_past_total_supply('ts2');
+}
+
+#[test]
+fn test_self_delegate() {
+    let mut state = setup_erc721_votes();
+    let contract_address = test_address();
+    let mut spy = spy_events();
+    start_cheat_caller_address(contract_address, OWNER());
+
+    state.delegate(OWNER());
+    spy.assert_event_delegate_changed(contract_address, OWNER(), ZERO(), OWNER());
+    spy.assert_only_event_delegate_votes_changed(contract_address, OWNER(), 0, 10);
+    assert_eq!(state.get_votes(OWNER()), 10);
+}
+
+#[test]
+fn test_delegate_to_recipient_updates_votes() {
+    let mut state = setup_erc721_votes();
+    let contract_address = test_address();
+    let mut spy = spy_events();
+    start_cheat_caller_address(contract_address, OWNER());
+
+    state.delegate(RECIPIENT());
+    spy.assert_event_delegate_changed(contract_address, OWNER(), ZERO(), RECIPIENT());
+    spy.assert_only_event_delegate_votes_changed(contract_address, RECIPIENT(), 0, 10);
+    assert_eq!(state.get_votes(RECIPIENT()), 10);
+    assert_eq!(state.get_votes(OWNER()), 0);
```

Comment:
I think we can improve the readability of the tests by using variables that better describe what they represent. For example, `delegator`, `delegatee`, and `total_votes` instead of `OWNER`, `RECIPIENT`, and `10`. This is especially helpful for longer test cases with event assertions. Forgive the nitpickiness 😅 I know this is a WIP